internal/controller/controllers/infraenv_controller.go

```diff
@@ -183,10 +184,46 @@ func (r *InfraEnvReconciler) updateInfraEnv(ctx context.Context, log logrus.Fiel
 
 	updateParams.InfraEnvUpdateParams.ImageType = r.Config.ImageType
 
+	existingKargs, err := kubeKernelArgs(internalInfraEnv)
+	if err != nil {
+		return nil, err
+	}
+	if !funk.Equal(infraEnv.Spec.KernelArguments, existingKargs) {
+		updateParams.InfraEnvUpdateParams.KernelArguments = internalKernelArgs(infraEnv.Spec.KernelArguments)
+	}
+
 	// UpdateInfraEnvInternal will generate an ISO only if there it was not generated before,
 	return r.Installer.UpdateInfraEnvInternal(ctx, updateParams, nil)
 }
 
+func kubeKernelArgs(internalInfraEnv *common.InfraEnv) ([]aiv1beta1.KernelArgument, error) {
+	var args models.KernelArguments
+	if internalInfraEnv.KernelArguments != nil {
```

Comment:
This is a good opportunity to also make sure `internalInfraEnv != nil`

Reply:
OK